### PR TITLE
Alisha/stock sidebar

### PR DESCRIPTION
**Pull request by awu-hub**
# Pull Request

## Description

Added in Stock-sidebar that uses tanstack query to take info from firestore. Changed design a bit so it's clearer as an overlay. 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

### Code Review Preparation
- [X] I have performed a self-review and reviewed the changed files
- [X] My code follows the project's style guidelines and passes linting
- [X] I have added comments in hard-to-understand areas and updated documentation

### Testing (Ignore for the first PR)
- [ ] I have added relevant tests and they pass locally
- [X] I have tested the changes manually, including edge cases

### Code Quality & Security
- [X] I have formatted my code and removed debugging/unused code
- [X] No TypeScript errors or linting issues
- [X] No sensitive information committed (API keys, passwords, etc.)
- [ ] User inputs are validated and sanitized

### Dependencies & Deployment
- [X] Dependencies are documented and package-lock.json is updated if needed
- [ ] Changes work with Firebase emulator and environment variables are documentedRetry

## Screenshots (If it is a front end feature screenshot is required)

image
image

## Additional Notes

I'm wondering if adding items of different categories is implemented yet or do we need to go directly into firebase to add because I had some trouble adding items of different categories on the website, I could only add items of category other so I haven't been able to test with multiple categories yet, but hypothetically it should still work through the map function. 

Also let me know if the maxCount function should be changed, because currently I don't think there's a field that contains that information, so for placeholder, I implemented the current maxCount function that just takes the maximum count out of all of the categories. 

## Summary by CodeRabbit

* **New Features**
  * Interactive Stock Sidebar with overlay, slide-in toggle, color-coded progress bars and percentage indicators per category.
  * Grid/table view toggle for warehouse list and preserved item create/edit/view/delete workflows.
  * Sidebar refetches data on open to show up-to-date category stock counts.

* **Refactor**
  * Layout reorganized to integrate the sidebar and conditional gallery/table rendering.

* **Chores**
  * Dependency version bump.

### CODE REVIEW

**Comment by awu-hub** [comment]
Changed the formatting of how the sidebar opens and uses CategoryAttributes type to retrieve information. 

image
image

File: app/inventory/warehouse/page.tsx [hunk 1]
```diff
@@ -136,142 +187,152 @@ export default function WarehousePage() {
...
+                            deleteItem={deleteItem}
+                        />
+                    )}
                 </div>
-            </div>
-            {isGridDisplay ? (
-                <WarehouseGallery
-                    inventoryRecords={items}
-                    openItem={setOpenedItem}
+                <StockSidebar
```
Comment (constrictingsnake) [inline app/inventory/warehouse/page.tsx]: The stock sidebar should be on a vast majority of the pages, not just the warehouse page (refer to figma)

**Review by constrictingsnake — COMMENTED** [review]
Left general comments about implementation and some changes

File: app/inventory/warehouse/page.tsx [hunk 2]
```diff
@@ -22,9 +27,11 @@ import { Timestamp } from "firebase/firestore";
 import { WarehouseGallery } from "@/components/inventory/WarehouseGallery";
 import { ItemViewModal } from "@/components/inventory/ItemViewModal";
 import { useCategories } from "@/lib/queries/categories";
+import { StockSidebar } from "@/components/inventory/StockSidebar";
 
 export default function WarehousePage() {
```
Comment (joelchem) [inline app/inventory/warehouse/page.tsx]: StockSidebar shouldn't change anything in warehouse page. It should live in the layout.tsx for warehouse.

**Review by joelchem — CHANGES_REQUESTED** [review]
Left some comments. Mostly general architectural changes about where code should go. There's also some other smaller styling/looks things. I think it makes the most sense to wait until your other PR is merged in so we can see how it interacts with the StockSidebar. Work on the review for that one before this one.
